Automattic/wp-calypso
# Site Assembler: Open Header, Sections, and Footer navigation to a side panel

_(edited on Aug 9 by @fushar)_

Related to:

- #78527

## Proposed Changes

* Show header, section, and footer patterns in a side panel
* Use an active state on the header/sections/footer buttons (same as the open category state)

### Before

https://github.com/Automattic/wp-calypso/assets/1525580/8616db67-5c92-4dc3-a4bc-d57bb2897718

### After

https://github.com/Automattic/wp-calypso/assets/1525580/6a3671c3-8574-4e44-bcea-48b83463bddd


## Testing Instructions



1. Access Site Assembler `/setup/with-theme-assembler/patternAssembler?siteSlug={ YOUR SITE }`
1. Click on "Header" and choose a header
1. Click on "Sections" and choose patterns (verify that the first selected category is All)
1. Click on "Footer" and choose a footer
1. Verify the active state of the header/sections/footer button
1. Verify the canvas scrolls to the header/footer area when the panel opens
1. Verify the selected header/section patterns/footer has a dark border on the pattern list (you'll have to move the focus from it)
1. Verify the hover, focus, and checked states of the header/sections/footer button on the main screen look as before
1. Verify that tracks events are not affected

## Pre-merge Checklist
